File: sample/src/main/java/com/github/lany192/box/sample/bean/Result.java
package com.github.lany192.box.sample.bean;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class Result<T> {
    private int code;
    private String msg;
    private T data;

    public boolean isSuccess() {
        return code == 200;
    }
}

File: sample/src/main/java/com/github/lany192/box/sample/bean/Result.kt
package com.github.lany192.box.sample.bean

class Result<T> {
    var code = 0
    var msg: String? = null

    var data: T? = null
        private set

    fun setData(data: T) {
        this.data = data
    }

    val isSuccess: Boolean
        get() = code == 200
}